src/client.ts

```diff
@@ -1,499 +1,424 @@
+      this.queryClient.request(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null) {
+            resolve(response.toObject())
+          }
+        },
+      )
+    })
   }
 
-  /**
-   * sendTxSyncMode
-   * @param data
-   */
-  async sendTxSyncMode(data: object): Promise<TransactionSyncMode> {
-    let response = await this.post('/txs', {
-      tx: data,
-      mode: 'sync',
+  async getReporters(validator: string): Promise<string[]> {
+    const request = new QueryReportersRequest()
+    request.setValidatorAddress(validator)
+    return new Promise((resolve, reject) => {
+      this.queryClient.reporters(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.getReporterList())
+          }
+        },
+      )
     })
+  }
 
-    let code = 0
-    let errorLog
-
-    if (response.code) {
-      code = parseInt(response.code)
-      errorLog = response.raw_log
-    }
+  async getLatestBlock(): Promise<GetLatestBlockResponse.AsObject> {
+    const request = new GetLatestBlockRequest()
+    return new Promise((resolve, reject) => {
+      this.serviceClient.getLatestBlock(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null) {
+            reject(err)
+            return
+          }
+
+          if (response !== null && response) {
+            resolve(response.toObject())
+          }
+        },
+      )
+    })
+  }
 
-    return {
-      txHash: Buffer.from(response.txhash, 'hex'),
-      code,
-      errorLog,
-    }
+  async getAccount(address: string): Promise<BaseAccount.AsObject> {
+    const request = new QueryAccountRequest()
+    request.setAddress(address)
+    return new Promise((resolve, reject) => {
+      this.authQueryClient.account(
+        request,
+        {} as grpc.Metadata,
+        (err, response) => {
+          if (err !== null || response === null) {
+            reject(err)
+            return
+          }
+          if (response.hasAccount()) {
+            let accBaseAccount = response
+              .getAccount()
+              .unpack(
+                BaseAccount.deserializeBinary,
+                'cosmos.auth.v1beta1.BaseAccount',
+              )
+            if (accBaseAccount.getAddress) {
+              resolve(accBaseAccount.toObject())
+            }
+          }
+          resolve(undefined)
+        },
+      )
+    })
   }
 
-  /**
-   * sendTxAsyncMode
-   * @param data
-   */
-  async sendTxAsyncMode(data: object): Promise<TransactionAsyncMode> {
-    let respose = await this.post('/txs', {
-      tx: data,
-      mode: 'async',
+  async getRequestIdByTxHash(txHash: string): Promise<any> {
```

Comment:
Return type should not be `any` type as the data is RequestID, which is number.

Reply:
Return type is now changed to array of string

Reply:
I think the function should return as a list of number,